transport-native-epoll/src/main/java/io/netty/channel/epoll/EpollEventLoop.java

```diff
@@ -201,7 +209,8 @@ private int epollWait(boolean oldWakenUp) throws IOException {
         long currentTimeNanos = System.nanoTime();
         long selectDeadLineNanos = currentTimeNanos + delayNanos(currentTimeNanos);
         for (;;) {
-            long timeoutMillis = (selectDeadLineNanos - currentTimeNanos + 500000L) / 1000000L;
+            long timeoutMillis = Math.min(MAX_EPOLL_WAIT_TIMEOUT,
```

Comment:
I'm trying to understand why we have a loop to call `Native.epollWait` in `EpollEventLoop` and we also loop in [netty_epoll_native_epollWait0](https://github.com/netty/netty/blob/4.1/transport-native-epoll/src/main/c/netty_epoll_native.c#L249) to ensure we wait for the desired time. `delayNanos` gets the amount of time until the "next schedule task to execute (max of 1 second) and scheduling a `Runnable` involves a queueing a wakeup task on the event loop [1]. So even if a new task is scheduled while we are waiting the event loop should be woken up to re-evaluate wait times. It seems like we could reduce thrashing on the event loop when there is a low workload and in general reduce code here. This was the approach I took in KQUEUE but I'm not sure if I'm missing something...

[1] https://github.com/netty/netty/blob/4.1/common/src/main/java/io/netty/util/concurrent/AbstractScheduledEventExecutor.java#L212